tools/squidclient/squidclient.cc

```diff
@@ -510,10 +508,22 @@ main(int argc, char *argv[])
         if (max_forwards > -1) {
             msg << "Max-Forwards: " << max_forwards << "\r\n";
         }
-        if (ProxyAuthorization.user)
-            ProxyAuthorization.commit(msg);
-        if (OriginAuthorization.user)
-            OriginAuthorization.commit(msg);
+        if (ProxyAuthorization.user) {
+            ProxyAuthorization.commit(msg); // handles missing passwords
+        } else if (ProxyAuthorization.password) {
+            std::cerr << "ERROR: Proxy authentication username is missing\n";
+            exit(EXIT_FAILURE);
+        }
+        if (OriginAuthorization.user) {
+            const auto savedPassword = OriginAuthorization.password;
+            if (pathPassword)
+                OriginAuthorization.password = pathPassword;
+            OriginAuthorization.commit(msg); // handles missing passwords
+            OriginAuthorization.password = savedPassword; // restore the global password setting
+        } else if (pathPassword || OriginAuthorization.password) {
+            std::cerr << "ERROR: WWW authentication username is missing\n";
```

Comment:
Please move the `pathPassword` part of this check into the `pathPassword`-setting code. That code is in a much better position to report (a URL-specific) failure. Something like `ERROR: Embedding a password in a cache manager command requires providing a username with -U: <actual mgr:command here>`. The moved code will still quit with `EXIT_FAILURE`, of course.

Reply:
Done in [PR1251](https://github.com/squid-cache/squid/pull/1251/).